README.md

```diff
@@ -1,304 +1,275 @@
+- Uploading and validation of location reference data
+- Deletion of locations where appropriate.
 
-    https://docs.gradle.org/current/userguide/checkstyle_plugin.html
+Most interactions with `pip-data-management` are performed through the API (specified in [API Documentation](#api-documentation)) either as a standalone service or via connections to other microservices.
 
-    Performs code style checks on Java source files using Checkstyle and generates reports from these checks.
-    The checks are included in gradle's *check* task (you can run them by executing `./gradlew check` command).
+## Features and Functionality
 
-  * pmd
+- Uploading/retrieval/deletion of publications into the service
+- Interfacing with local or hosted Postgres instances for metadata and retrieval and Azure Blob Storage for raw files.
+- Parsing and validation of ingested json files.
+- Flyway for database modifications via SQL ingestion.
+- Secure/Insecure Mode: Use of bearer tokens for authentication with the secure instance (if desired)
+- Azure Blob Storage: Handles interactions with the CaTH Azure Blob Storage instance (or local Azurite emulator/Azure Storage Explorer instances)
+- Scheduled cronjobs for daily tasks (e.g. retention period checks for archival purposes)
+- OpenAPI Spec/Swagger-UI: Documents and allows users or developers to access API resources within the browser.
+- Integration tests using TestContainers for dummy database operations.
+
+## Architecture Diagram
 
-    https://docs.gradle.org/current/userguide/pmd_plugin.html
+![Architecture Diagram for pip-data-management](./data-man-arch.png)
+
+The above diagram is somewhat simplified for readability (e.g. it does not include secure/insecure communications, but those are covered elsewhere).
 
-    Performs static code analysis to finds common programming flaws. Included in gradle `check` task.
+## Getting Started
 
+### Prerequisites
 
-  * jacoco
+##### General
 
-    https://docs.gradle.org/current/userguide/jacoco_plugin.html
+- [Java JDK 17](https://openjdk.org/projects/jdk/17/) - this is used throughout all of our services.
+- REST client of some description (e.g. [Curl](https://github.com/curl/curl), [Insomnia](https://insomnia.rest/), [Postman](https://www.postman.com/)). Swagger-UI can also be used to send requests.
+- Docker - used to run integration tests due to our use of [TestContainers](https://www.testcontainers.org/)
 
-    Provides code coverage metrics for Java code via integration with JaCoCo.
-    You can create the report by running the following command:
+##### Local development
 
-    ```bash
-      ./gradlew jacocoTestReport
-    ```
+- [Azurite](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azurite) - Local Azure emulator used along with Azure Storage explorer for local storage.
+- [Azure Storage Explorer](https://azure.microsoft.com/en-us/products/storage/storage-explorer) - Used for viewing and storing blobs within an Azurite instance locally.
```

Comment:
I think we should add Postgres in here too

Reply:
resolved